src/storage/media-search-store.js

```diff
@@ -113,33 +115,68 @@ export default class MediaSearchStore extends EventTarget {
     pushHistoryPath(this.history, location.pathname, searchParams.toString());
   };
 
-  getSearchClearedSearchParams = location => {
+  getSearchClearedSearchParams = (location, keepSource) => {
     const searchParams = new URLSearchParams(location.search);
 
     // Strip browsing query params
     searchParams.delete("q");
     searchParams.delete("filter");
     searchParams.delete("cursor");
-    searchParams.delete("media_source");
     searchParams.delete("media_nav");
 
+    if (!keepSource) {
+      searchParams.delete("media_source");
+    }
+
     return searchParams;
   };
 
+  stashLastSearchParams = location => {
+    const searchParams = new URLSearchParams(location.search);
+    this._stashedParams = {};
+    this._stashedSource = this.getUrlMediaSource(location);
+
+    for (const param of SEARCH_CONTEXT_PARAMS) {
+      const value = searchParams.get(param);
+
+      if (value) {
+        this._stashedParams[param] = value;
+      }
+    }
+  };
+
   sourceNavigateToDefaultSource = () => {
-    this.sourceNavigate(SOURCES[0]);
+    this.sourceNavigate(this._stashedSource ? this._stashedSource : SOURCES[0], false, true);
   };
 
-  sourceNavigate = (source, hideNav) => {
+  sourceNavigate = (source, hideNav, useLastStashedParams) => {
```

Comment:
Maybe make this an internal `_sourceNavigate` method and make public `sourceNavigateHiddenNavWithStash` and `sourceNavigateWithNav` methods or use a `params` object.